### PR TITLE
Login Lambda

### PR DESCRIPTION
**Pull request by jaredcurtis717**
Login Lambda written. SES functionality started, currently not working.

### CODE REVIEW

**Comment by jaredcurtis717** [comment]
Note: This pull request has the login lambda not return anything other than that the process succeeded. Currently in my local branch it has been changed to return the user's ID

File: backend/src/lambda/login.ts [hunk 1]
```diff
@@ -14,11 +18,18 @@ async function handler(event: APIGatewayEvent, context: Context) {
 
   // search for contact in users table
   // if contact doesn't exist, create an entry
+  const userId = await getUserIdByContactString(contactString as string)
```
Comment (nolanbarry) [inline backend/src/lambda/login.ts:21]: contactString is already a string and doesn't need to be typecasted

File: backend/src/lambda/login.ts [hunk 2]
```diff
@@ -14,11 +18,18 @@ async function handler(event: APIGatewayEvent, context: Context) {
...
+
   // create auth table entry with new otp and user id
-  // text/email otp to user
-  // return 200 OK {success: true}
+  const otp = await login(userId)
+
+  // text/email otp to user 
+  //TODO: allow for email or text based on mode of contact.
+  // Currently this will always try to send an email to the contactString, even if it is a phone number
+  sendMessage(otp as string, "Secret Santa One Time Password", contactString)
```
Comment (nolanbarry) [inline backend/src/lambda/login.ts:29]: `otp` also doesn't need a typecast, typescript can infer the type based on the return type of `login`

File: backend/src/lambda/login.ts [hunk 3]
```diff
@@ -14,11 +18,18 @@ async function handler(event: APIGatewayEvent, context: Context) {
...
 
-  return response(200, { message: "Function not implemented." })
+  // return 200 OK {success: true}
+  return response(200, { success: true })
```
Comment (nolanbarry) [inline backend/src/lambda/login.ts:32]: should return user id, right?

File: backend/src/services/ses.ts [hunk 4]
```diff
@@ -1,9 +1,51 @@
...
+  sendEmail({
+    emails: emails,
+    body: getMessageHtml(message),
+    subject: subject, //This needs to change based on what kind of message we're sending
+    source: constants.sesSource
+  })
+}
+
+function getMessageHtml(message:string){
+  //Should probably move this to a .html file and read it instead
```
Comment (nolanbarry) [inline backend/src/services/ses.ts:19]: I think this is fine

**Review by nolanbarry — COMMENTED** [review]
Not sure if you should be returning a user id or not

**Pull request merged**